pandas/tests/dtypes/test_common.py

```diff
@@ -754,3 +756,12 @@ def test_validate_allhashable():
 
     with pytest.raises(TypeError, match="list must be a hashable type"):
         com.validate_all_hashable([], error_name="list")
+
+
+def test_pandas_dtype_numpy_warning():
+    # GH#51523
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
can you do tm.assert_produces_warning() instead?

Reply:
sure thing, I've updated the testcase.